sw/device/lib/testing/edn_testutils.h

```diff
@@ -29,4 +30,15 @@ dif_edn_auto_params_t edn_testutils_auto_params_build(bool disable_rand,
                                                       unsigned int res_itval,
                                                       unsigned int glen_val);
 
+/**
+ * Returns status_ok if the expected state has been reached and
+ * deadline_exceeded if the timeout has been reached.
```

Comment:
I think the function should indeed have a deadline / timeout but can't see this in the implementation. Is it part of the `TRY()` macro or did you forget to add it?

Reply:
Yes thanks, I forgot. I changed the implementation of the wait function a little bit. It should now include a timeout.